pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/impl/RASPassPortService.java

```diff
@@ -0,0 +1,210 @@
+                 EXPIRED_POLLING -> handleFailedValidationResponse(response, user);
+        };
+    }
+
+    /**
+     * If a passport is anything but VALID the user will be logged out and their passport be cleared.
+     *
+     * @param validateResponse The response from the passport validation
+     * @param user             The user to log out
+     * @return true if the user was successfully updated
+     */
+    private boolean handleFailedValidationResponse(String validateResponse, User user) {
+        user.setPassport(null);
+        this.userService.save(user);
+        this.userService.logoutUser(user);
+        this.logger.info("handleFailedValidationResponse - {} - USER LOGGED OUT - {}", validateResponse, user.getSubject());
+        return false;
+    }
+
+    private boolean handleValidValidationResponse(String validateResponse, User user) {
+        this.logger.info("handleValidValidationResponse PASSPORT VALIDATE RESPONSE ___ {} ___ USER {}", validateResponse, user.getSubject());
+        return true;
+    }
+
+    public Optional<String> validateVisa(String visa) {
+        if (StringUtils.isBlank(visa)) {
+            logger.error("validatePassport() VISA IS EMPTY");
+            return Optional.empty();
+        }
+
+        MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+        queryParams.add("visa", visa);
+        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(queryParams, null);
+
+        String responseVal = PassportValidationResponse.VISA_EXPIRED.getValue();
+        ResponseEntity<String> resp;
+        try {
+            resp = this.restClientUtil.retrievePostResponse(this.rasURI + "/passport/validate", request);
+            responseVal = resp.getBody();
+        } catch (Exception e) {
+            logger.error("validatePassport() FAILED TO VALIDATE VISA ___ {}", e.getMessage());
+        }
+
+        return Optional.ofNullable(responseVal);
+    }
+
+    public Set<RasDbgapPermission> ga4gpPassportToRasDbgapPermissions(List<String> ga4ghPassports) {
+        if (ga4ghPassports == null) {
+            return null;
+        }
+
+        logger.debug("Converting ga4ghPassports to RasDbgapPermissions");
+        HashSet<RasDbgapPermission> rasDbgapPermissions = new HashSet<>();
+        ga4ghPassports.forEach(ga4ghPassport -> {
+            Optional<Ga4ghPassportV1> parsedGa4ghPassportV1 = JWTUtil.parseGa4ghPassportV1(ga4ghPassport);
+            if (parsedGa4ghPassportV1.isPresent()) {
+                Ga4ghPassportV1 ga4ghPassportV1 = parsedGa4ghPassportV1.get();
+                logger.info("ga4gh_passport_v1: {}", ga4ghPassportV1);
+
+                rasDbgapPermissions.addAll(ga4ghPassportV1.getRasDbgagPermissions());
+            }
+        });
+
+        return rasDbgapPermissions;
+    }
+
+    public void setRasURI(String rasURI) {
+        this.rasURI = rasURI;
```

Comment:
Do we not want to remove trailing slashes here too?

Reply:
I can include it. I didn't think it would be necessary because it is only used for unit test.